pkg/pool-weighted/contracts/managed/ManagedPoolStorageLib.sol

```diff
@@ -0,0 +1,156 @@
+    uint256 private constant _SWAP_ENABLED_OFFSET = 254;
+    uint256 private constant _MUST_ALLOWLIST_LPS_OFFSET = 255;
+
+    uint256 private constant _TIMESTAMP_WIDTH = 32;
+    uint256 private constant _SWAP_FEE_START_PERCENTAGE_WIDTH = 64;
+    uint256 private constant _SWAP_FEE_END_PERCENTAGE_WIDTH = 62;
+
+    // Getters
+
+    /**
+     * @notice Returns whether the Pool currently allows swaps (and by extension, non-proportional joins/exits).
+     */
+    function getSwapsEnabled(bytes32 miscData) internal pure returns (bool) {
+        return miscData.decodeBool(_SWAP_ENABLED_OFFSET);
+    }
+
+    /**
+     * @notice Returns whether addresses must be allowlisted to add liquidity to the Pool.
+     */
+    function getLPAllowlistEnabled(bytes32 miscData) internal pure returns (bool) {
+        return miscData.decodeBool(_MUST_ALLOWLIST_LPS_OFFSET);
+    }
+
+    /**
+     * @notice Returns the percentage progress through the current gradual weight change.
+     */
+    function getGradualWeightChangeProgress(bytes32 miscData) internal view returns (uint256) {
+        (uint256 startTime, uint256 endTime) = _getWeightChangeFields(miscData);
+
+        return GradualValueChange.calculateValueChangeProgress(startTime, endTime);
+    }
+
+    /**
+     * @notice Returns the current value of the swap fee percentage.
+     * @dev Computes the current swap fee percentage, which can change every block if a gradual swap fee
+     * update is in progress.
+     */
+    function getSwapFeePercentage(bytes32 miscData) internal view returns (uint256) {
+        (
+            uint256 startTime,
+            uint256 endTime,
+            uint256 startSwapFeePercentage,
+            uint256 endSwapFeePercentage
+        ) = _getSwapFeeFields(miscData);
+
+        return
+            GradualValueChange.getInterpolatedValue(startSwapFeePercentage, endSwapFeePercentage, startTime, endTime);
+    }
+
+    // Setters
+
+    function setSwapsEnabled(bytes32 miscData, bool enabled) internal pure returns (bytes32) {
+        return miscData.insertBool(enabled, _SWAP_ENABLED_OFFSET);
+    }
+
+    function setLPAllowlistEnabled(bytes32 miscData, bool enabled) internal pure returns (bytes32) {
+        return miscData.insertBool(enabled, _MUST_ALLOWLIST_LPS_OFFSET);
+    }
+
+    function setWeightChangeData(
+        bytes32 miscData,
+        uint256 startTime,
+        uint256 endTime
+    ) internal pure returns (bytes32) {
+        return
+            miscData.insertUint(startTime, _WEIGHT_START_TIME_OFFSET, _TIMESTAMP_WIDTH).insertUint(
+                endTime,
+                _WEIGHT_END_TIME_OFFSET,
+                _TIMESTAMP_WIDTH
+            );
+    }
+
+    function setSwapFeeData(
+        bytes32 miscData,
+        uint256 startTime,
+        uint256 endTime,
+        uint256 startSwapFeePercentage,
+        uint256 endSwapFeePercentage
+    ) internal pure returns (bytes32) {
+        // Add scope to prevent "Stack too deep"
```

Comment:
Do we really need this? That's a bit surprising - I imagine it has to do with the temporary values that result from the chained `.insert` calls.

It'd be interesting to change this to alternative forms and see if we get a noticeable impact on e.g bytecode size, say
```
data = data.insert(...)
data = data.insert(...)
data = data.insert(...)
return data
```

Reply:
The bytecode cause by the two approaches seems to be identical.